### PR TITLE
solveset with piecewise: issue 19718

### LINKED ISSUE #19718: solveset and Piecewise give wrong result
Please consider this:
```
from sympy.abc import x
from sympy import Piecewise
from sympy import Intersection
from sympy import solveset
from sympy import S

g = Piecewise( (1, x > 10), (0, x <= 10) )

#prints 0 as expected
print g.subs(x, 10)

# prints 1 as expected
print g.subs(x, 11)

# should print sth like Interval.open(10, infty)
print solveset(g > 0 , x, S.Reals)
```
The result is EmptySet for me (python 2.7.16 and sympy 1.6.1)

### PR DESCRIPTION
**Pull request by maurogaravello**
#### References to other Issues or PRs

Fixes #19718 

#### Brief description of what is fixed or changed

`Solveset` with `inequality` and `Piecewise` does not work properly.
Indeed it is `_solveset` which returns a wrong result when an argument is a `BooleanTrue`.
Now, when the argument is a `BooleanTrue`, it returns the domain.

A test is added.

#### Release Notes

- solvers
  - Handling of inequalities involving Piecewise is improved in solveset.

### CODE REVIEW

File: sympy/solvers/solveset.py [hunk 1]
```diff
@@ -969,6 +969,10 @@ def _solveset(f, symbol, domain, _check=False):
     given symbol."""
     # _check controls whether the answer is checked or not
     from sympy.simplify.simplify import signsimp
+    from sympy.logic.boolalg import BooleanTrue
+
+    if isinstance(f, BooleanTrue):
+        return domain
```
Comment (czgdp1807) [inline sympy/solvers/solveset.py:975]: This line isn't covered by the tests according to [this report](https://codecov.io/gh/sympy/sympy/pull/19986/diff). Can you please add some tests?

Comment (maurogaravello) [reply sympy/solvers/solveset.py:975]: Do you mean something like this?
```
    f = BooleanTrue()
    assert solveset(f, x, domain=Interval(-3, 10)) == Interval(-3, 10)
```

Comment (czgdp1807) [reply sympy/solvers/solveset.py:975]: Yes. This should work.

**Comment by czgdp1807** [comment]
@smichr Is this good to go?

**Comment by oscarbenjamin** [comment]
Looks good. Thanks!

**Pull request merged**